src/client/Microsoft.Identity.Client/Internal/Requests/ManagedIdentityAuthRequest.cs

```diff
@@ -60,6 +60,18 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
             MsalAccessTokenCacheItem cachedAccessTokenItem = await GetCachedAccessTokenAsync()
                 .ConfigureAwait(false);
 
+            // If using IMDSv2 (mTLS) and the mTLS cert is near expiry, skip the AT cache
+            if (cachedAccessTokenItem != null && ManagedIdentityClient.s_sourceName == ManagedIdentitySource.ImdsV2)
+            {
+                string identityKey = ServiceBundle.Config.ClientId;
+
+                if (ManagedIdentityClient.IsMtlsCertExpiringSoon(identityKey))
```

Comment:
How about smth like `if (cert = cache.GetCertifiacate(key) && ~cert.IsExpiringSoon())`? This will avoid having to expose a new complex API that does 2 things at once.

Alternatively, the cache.GetCertificate() can simply return NULL if the certificate is close to expiry. This is what we do for tokens.